crates/terminal_view/src/terminal_view.rs

```diff
@@ -187,11 +187,12 @@ impl TerminalView {
             Event::Open(maybe_navigation_target) => match maybe_navigation_target {
                 MaybeNavigationTarget::Url(url) => cx.open_url(url),
 
-                MaybeNavigationTarget::PathLike(maybe_path) => {
+                MaybeNavigationTarget::PathLike(maybe_path, cwd) => {
                     if !this.can_navigate_to_selected_word {
                         return;
                     }
-                    let potential_abs_paths = possible_open_targets(&workspace, maybe_path, cx);
+                    let potential_abs_paths =
+                        possible_open_targets(&workspace, cwd, maybe_path, cx);
                     if let Some(path) = potential_abs_paths.into_iter().next() {
```

Comment:
This seems to be my mistake (sorry), the code looks odd: it takes only the first `worktree.read(cx).abs_path().join(&maybe_path)` and checks it for existence.

First, we should do that for every path that's given already and second, the new code makes it even more dangerous: it forces us to only check the `cwd + maybe_path` join and avoid all worktrees (which might be a bad thing, if the relative path is only inside one of the later worktrees)